internal/cloud/realm/app.go

```diff
@@ -54,10 +54,10 @@ func (c *client) CreateApp(groupID, name string, meta AppMeta) (App, error) {
 	if resErr != nil {
 		return App{}, resErr
 	}
-	defer res.Body.Close()
 	if res.StatusCode != http.StatusCreated {
-		return App{}, parseResponseError(res)
+		return App{}, api.ErrUnexpectedStatusCode{"create app", res.StatusCode}
 	}
+	defer res.Body.Close()
```

Comment:
why did these defers move?

Reply:
with the new `api.ErrUnexpectedStatusCode`, we don't attempt to read from the `res.Body` until after we've determined we received the expected status code (previously, `parseResponseError` did read from `res.Body`)

the was brought about by handling automatic auth refresh in the client api calls.  we only return `res` from `client.do` when its status code is between 200 and 299.  so "parsing" the response for a server error was never really going to work in that scenario anyways (it wasn't a server error, just an unexpected 2XX code)